### PR TITLE
IEN-950 | Undo-EoJ: Implement Logic for Re-Engage milestone

### PR DESCRIPTION
**Pull request by jerry-ey**
NOTED: This is merging to branch `IEN-941` as it's continuing from re-engaged UI

- add 3 events: Create, Update, Delete re-engaged milestone
- These 3 events will trigger same service function which would go through complete_check and incomplete_check
- Fixed PR feedback from branch 941

### CODE REVIEW

File: apps/api/src/applicant/endofjourney.service.ts [hunk 1]
```diff
@@ -93,8 +96,8 @@ export class EndOfJourneyService implements OnModuleInit {
     }
   }
 
-  async handleEndOfJourney<T>(
-    getter: Getter<T>,
+  async handleEndOfJourney<T, U = any>(
```
Comment (dbayly-freshworks) [inline apps/api/src/applicant/endofjourney.service.ts:99]: Any chance we can specify a type here?

File: apps/api/src/applicant/endofjourney.service.ts [hunk 2]
```diff
@@ -214,20 +245,176 @@ export class EndOfJourneyService implements OnModuleInit {
...
+        'status.status AS milestone', // Include milestone (status)
+        'audit.start_date AS milestone_start_date', // Include milestone start date
+      ])
+      .leftJoin('ien_applicant_status_audit', 'audit', 'audit.applicant_id = applicant.id')
+      .leftJoin('ien_applicant_status', 'status', 'status.id = audit.status_id')
+      .where('applicant.id = :id', { id: payload.applicant.id });
+
+    const rawResults = await query.getRawMany();
+    const applicants = rawResults.reduce((result, row) => {
+      let applicant = result.find((a: any) => a.applicant_id === row.applicant_id);
```
Comment (dbayly-freshworks) [inline apps/api/src/applicant/endofjourney.service.ts:374]: Same here

**Review by dbayly-freshworks — APPROVED** [review]
Looks good to me, two small nits

File: apps/api/src/applicant/endofjourney.service.ts (hunk 1 above)
Comment (jerry-ey) [reply apps/api/src/applicant/endofjourney.service.ts]: Thank you, fixed!

File: apps/api/src/applicant/endofjourney.service.ts (hunk 2 above)
Comment (jerry-ey) [reply apps/api/src/applicant/endofjourney.service.ts]: Thank you, fixed!

**Pull request merged**